numpy/core/numeric.py

```diff
@@ -662,14 +662,21 @@ def flatnonzero(a):
     return np.nonzero(np.ravel(a))[0]
 
 
-_mode_from_name_dict = {'v': 0,
-                        's': 1,
-                        'f': 2}
+_mode_from_name_dict = {'valid': 0,
+                        'same': 1,
+                        'full': 2}
 
 
-def _mode_from_name(mode):
+def _mode_from_name(mode, default_mode):
     if isinstance(mode, str):
-        return _mode_from_name_dict[mode.lower()[0]]
+        if mode.lower() not in _mode_from_name_dict.keys():
```

Comment:
Instead of a double lookup, I'd recommend:
```python
try:
    return _mode_from_name_dict[mode.lower()]
except KeyError:
    pass
# deprecation handling goes here
```

Reply:
Try/except clause makes more sense, I'll try this 👍🏼